sources/imdsclient/src/lib.rs

```diff
@@ -21,20 +21,28 @@ use log::{debug, info, trace, warn};
 use reqwest::Client;
 use serde_json::Value;
 use snafu::{ensure, OptionExt, ResultExt};
-use std::time::Duration;
-use tokio::time;
+use tokio::time::{sleep, timeout, Duration};
+use tokio_retry::{strategy::FibonacciBackoff, Retry};
 
 const BASE_URI: &str = "http://169.254.169.254";
 const PINNED_SCHEMA: &str = "2021-01-03";
 
 // Currently only able to get fetch session tokens from `latest`
 const SESSION_TARGET: &str = "latest/api/token";
 
+fn retry_token_strategy() -> impl Iterator<Item = Duration> {
+    // Retries at 0.5s, 1s, 1.5s, 2.5s, 4s, 6.5s, 10.5s, 17s, and then every 10s after.
+    FibonacciBackoff::from_millis(500)
+        .factor(1)
```

Comment:
```suggestion
```
I think specifying the default here just raises questions, and I'd argue that the method doesn't really make much sense anyway, so they're not super useful questions to raise...